src/utils.js

```diff
@@ -312,10 +312,109 @@ export function evalSetCurrentStory({ waitFor }, story) {
 
     // resolve when rendered, reject on any other renderer event
     return new Promise((resolve, reject) => {
-      channel.on('storyRendered', resolve);
       channel.on('storyMissing', (err) => reject(err || new Error('Story Missing')));
       channel.on('storyErrored', (err) => reject(err || new Error('Story Errored')));
       channel.on('storyThrewException', (err) => reject(err || new Error('Story Threw Exception')));
+
+      // Handle story rendered, but also check for loaders
+      channel.on('storyRendered', () => {
+        // After the story is rendered, add a small delay before checking loaders
+        // This helps ensure that any post-render loader state changes have time to occur
+        setTimeout(() => {
+          // Wait for any loaders to disappear before resolving
+          waitForLoadersToDisappear().then(resolve).catch(reject);
+        }, 100);
+      });
     });
+
+    // Helper function to wait for all loaders to disappear
+    function waitForLoadersToDisappear() {
+      return new Promise((resolve, reject) => {
+        const timeout = setTimeout(() => {
+          // Collect information about loaders that are still visible for debugging
+          const initialLoader = document.getElementById('preview-loader');
+          const preparingStory = document.querySelector('.sb-preparing-story');
+          const preparingDocs = document.querySelector('.sb-preparing-docs');
+          const bodyClasses = document.body.classList.toString();
+
+          let loaderInfo = {};
+          if (initialLoader) loaderInfo.initialLoader = true;
+          if (preparingStory) loaderInfo.preparingStory = !!window.getComputedStyle(preparingStory).display !== 'none';
+          if (preparingDocs) loaderInfo.preparingDocs = !!window.getComputedStyle(preparingDocs).display !== 'none';
+          loaderInfo.bodyClasses = bodyClasses;
+
+          reject(new Error(`Timed out waiting for Storybook loaders to disappear: ${JSON.stringify(loaderInfo)}`));
```

Comment:
we shouldnt be rejecting right ? we want to print warning and take snapshot anyway after timeout